File: sol/717/717.py
class Solution:
    def isOneBitCharacter(self, bits: List[int]) -> bool:
        n = len(bits)
        ptr = 0
        lastIsSingle = False
        while ptr < n:
            if bits[ptr] == 0:
                lastIsSingle = True
                ptr += 1
            else:
                lastIsSingle = False
                ptr += 2
        return lastIsSingle

File: sol/717/717.cpp
struct Solution {
    bool isOneBitCharacter(vector<int>& bits) {
        int n = bits.size(), ptr = 0;
        bool isSingle = false;
        while (ptr < n){
            if (bits[ptr] == 0){
                isSingle = true;
                ++ptr;
            } else {
                isSingle = false;
                ++++ptr;
            }
        }
        return isSingle;
    }
};
